src/bin/spin.rs

```diff
@@ -35,18 +36,24 @@ async fn main() -> Result<(), Error> {
 
 lazy_static! {
     pub static ref VERSION: String = build_info();
+    pub static ref VERSION_VERBOSE: String = build_info_verbose();
 }
 
 /// Helper for passing VERSION to structopt.
 fn version() -> &'static str {
     &VERSION
 }
 
+fn version_verbose() -> &'static str {
+    &VERSION_VERBOSE
+}
+
 /// The Spin CLI
 #[derive(Parser)]
 #[clap(
     name = "spin",
     version = version(),
+    long_version = version_verbose(),
```

Comment:
We've been leading people to use `spin --version` in the docs, and this will now make that print a significant amount of extra info that most people shouldn't care about (and it had quite a bit of that in it already!).  Is there another way we could surface the long form?

(I am also wondering if, longer term, we should default it to _just_ showing the version, and having the commit SHA and date appear only in the long form.  But that is definitely out of scope for this PR!)

Reply:
I would prefer -V -v as well, but I guess introducing a verbose global variable could create further discussions and fighting with clap v3. If changing long_version output is too disruptive, I can explore -V -v or leave it out for now.